txn/mgo_test.go

```diff
@@ -17,16 +21,39 @@ type MgoSuite struct {
 	session *mgo.Session
 }
 
-var mgoaddr = "127.0.0.1:50017"
+const mgoip = "127.0.0.1"
```

Comment:
You're right - these changes are only tangentially related to the upsert fix. Would you rather I split them (and the travis config change) out into a separate PR? It would need to get merged after this one, otherwise the txn tests will be failing.

 You're right, dbtest handles this much more nicely. I'll rewrite txn_test.go to use dbtest and remove mgo_tes.go.